### ISSUE TITLE
[Proposal] add new types of scopes

### ISSUE THREAD

**Issue by lk77** [yajra/laravel-datatables-buttons#30]
Hello,

i think it would be useful to add another type of scopes, currently, it's used to add a condition to the query,
but i think we could have scopes for other things that the query. For example we could have a CheckboxHtmlScope, to add checkboxes when we need to, without having to do if statements in DataTables to know if checkboxes needs to be enabled or not.
we could have a SkipPagingAjaxScope for example to skip pagination when we need to, on any datatable we want.

to summarize, i think we could have :

- QueryScopes
- AjaxScopes
- HtmlScopes

and in the datatable,
we could have 
```
return $this->applyQueryScopes($query);
return $this->applyAjaxScopes($datatable);
return $this->applyHtmlScopes($builder);
```
for example in Datatable.

what do you think ?

**Comment by yajra** [comment]
Sorry for late reply. The idea looks good. Can you please submit a PR for POC? I may not be able to do some updates atm. Thanks!
